### PR TITLE
Routing Home Task + PR fixes

### PR DESCRIPTION
**Pull request by rkostik**

### CODE REVIEW

File: src/components/Header/Header.jsx [hunk 1]
```diff
@@ -3,10 +3,12 @@ import Logo from "./Logo/Logo.jsx";
 import styles from './Header.module.css';
 import BtnAdd from "./BtnAdd/BtnAdd.jsx";
 import Search from "./Search/Search.jsx";
-import PropTypes from "prop-types";
+import PropTypes from 'prop-types';
+import {useParams} from 'react-router-dom';
```
Comment (artem-blazhko) [inline src/components/Header/Header.jsx:7]: I see that useParams is not used in this file

File: src/components/Header/HeaderContent.jsx [hunk 2]
```diff
@@ -1,17 +1,27 @@
-import React from 'react';
+import React, {useEffect} from 'react';
 import Header from './Header.jsx';
 import HeaderDetail from './HeaderDetail/HeaderDetail.jsx';
 import {useSelector} from 'react-redux';
-
+import {useHistory, useLocation, useParams} from "react-router-dom";
```
Comment (artem-blazhko) [inline src/components/Header/HeaderContent.jsx:5]: useHistory is not used in this file

File: src/components/HistoryListener.jsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    // if is the first time render, show loading
+/*    if (isFirstRender) {
+        return <p>Loading...</p>;
+    }*/
+
+    return children;
+};
+
+export default HistoryListener;
```
Comment (artem-blazhko) [inline src/components/HistoryListener.jsx:22]: Is it used somewhere?

File: src/components/Header/HeaderContent.jsx (hunk 2 above)
Comment (rkostik) [reply src/components/Header/HeaderContent.jsx:5]: Fixed

File: src/components/HistoryListener.jsx (hunk 3 above)
Comment (rkostik) [reply src/components/HistoryListener.jsx:22]: Fixed

File: src/components/Header/Header.jsx (hunk 1 above)
Comment (rkostik) [reply src/components/Header/Header.jsx:7]: Fixed

**Pull request merged**